src/usr/local/www/system_camanager.php

```diff
@@ -99,6 +99,7 @@
 	$pconfig['serial'] = $a_ca[$id]['serial'];
 	if (!empty($a_ca[$id]['prv'])) {
 		$pconfig['key'] = base64_decode($a_ca[$id]['prv']);
+		$pconfig['encryptkey'] = (strpos($pconfig['key'], 'ENCRYPTED') !== false);
```

Comment:
I don't know if that's the right way to check for encrypted keys. It was this way before, but there's the very slim chance that the base64 encoded string might contain `ENCRYPTED`. Maybe this should be a function in `certs.inc`?

Reply:
IIRC you can try the openssl function `openssl_pkey_get_private()` on the key with an empty password and if it fails/returns empty then it's safe to assume that it's encrypted. I haven't tested that so I'm not sure exactly what/how it returns, but that should be reliable.

Reply:
I'll try that, and I'll factor the check out into a separate function in any case.